example/json-body/json_body.hpp

```diff
@@ -0,0 +1,177 @@
+//
+// Copyright (c) 2022 Seth Heeren (sgheeren at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef BOOST_BEAST_EXAMPLE_JSON_BODY_JSON_BODY_HPP
+#define BOOST_BEAST_EXAMPLE_JSON_BODY_JSON_BODY_HPP
+
+#include <algorithm>
+//#include <boost/beast/core/buffer_traits.hpp>
+//#include <boost/beast/core/detail/buffer.hpp>
+//#include <boost/beast/core/detail/clamp.hpp>
+//#include <boost/beast/core/detail/config.hpp>
+#include <boost/asio/buffer.hpp>
+#include <boost/asio/buffers_iterator.hpp>
+#include <boost/beast/http/error.hpp>
+#include <boost/beast/http/message.hpp>
+#include <boost/json.hpp>
+#include <boost/optional.hpp>
+#include <cstdint>
+#include <utility>
+
+namespace boost {
+namespace beast {
+namespace contrib {
+
+/** A <em>Body</em> using a `boost::json::value`
+
+    This body uses a `boost::json::value` as a memory-based container for
+    holding message payloads. Messages using this body type may be serialized
+    and parsed.
+*/
+struct json_body
+{
+    /** The type of container used for the body
+
+        This determines the type of @ref message::body
+        when this body type is used with a message container.
+    */
+    using value_type = boost::json::value;
+
+    /** Returns the payload size of the body
+
+        When this body is used with @ref message::prepare_payload,
+        the Content-Length will be set to the payload size, and
+        any chunked Transfer-Encoding will be removed.
+    */
+    static std::uint64_t
+    size(value_type const& v)
```

Comment:
oh no... we don't want to have the fixed size. JSON bodies should always use chunked.